simd/unit_tests/include/TestSIMD_MathOps.hpp

```diff
@@ -29,21 +29,34 @@ void host_check_math_op_one_loader(BinaryOp binary_op, std::size_t n,
   for (std::size_t i = 0; i < n; i += width) {
     std::size_t const nremaining = n - i;
     std::size_t const nlanes     = Kokkos::min(nremaining, width);
-    simd_type first_arg;
+    if ((std::is_same_v<BinaryOp, divides> ||
+         std::is_same_v<BinaryOp, divides_eq>)&&nremaining < width)
+      continue;
+    simd_type first_arg(zero_init<simd_type>());
     bool const loaded_first_arg =
         loader.host_load(first_args + i, nlanes, first_arg);
-    simd_type second_arg;
+    simd_type second_arg(zero_init<simd_type>());
     bool const loaded_second_arg =
         loader.host_load(second_args + i, nlanes, second_arg);
     if (!(loaded_first_arg && loaded_second_arg)) continue;
-    simd_type expected_result;
+    simd_type expected_result(zero_init<simd_type>());
+
+    // gcc build with cxxflag of -g and -O2 or above doesn't seem to properly
+    // load values into simd vectors until simd values are directly accessed.
+    // Placing a memory fence to ensure that simd values are fully loaded
+    // before executing simd instructions.
+#if defined(KOKKOS_COMPILER_GNU) && defined(NDEBUG)
+    __sync_synchronize();
+#endif
```

Comment:
That issue should go away if we don't try to access elements via `reinterpret_cast`, right?

Reply:
Yes, these memory fences will be deleted once the reinterpret cast issue is fixed.